test/integration/testHTAP.js

```diff
@@ -0,0 +1,102 @@
+const assert = require('assert');
+const async = require('async');
+const connOption = require('./connectionOptions').valid;
+const testUtil = require('./testUtil');
+
+describe('Query Context Cache test', function () {
+  let connection;
+  beforeEach(() => {
+    connection = testUtil.createConnection(connOption);
+  });
+  const querySet = [
+    {
+      sqlTexts:[
+        'create or replace database db1',
+        'create or replace hybrid table t1 (a int primary key, b int)',
+        'insert into t1 values (1, 2), (2, 3), (3, 4)'
+      ],
+      QccSize:2,
+    },
+    {
+      sqlTexts:[
+        'create or replace database db2',
+        'create or replace hybrid table t2 (a int primary key, b int)',
+        'insert into t2 values (1, 2), (2, 3), (3, 4)'
+      ],
+      QccSize:3,
+    },
+    {
+      sqlTexts:[
+        'create or replace database db3',
+        'create or replace hybrid table t3 (a int primary key, b int)',
+        'insert into t3 values (1, 2), (2, 3), (3, 4)'
+      ],
+      QccSize:4,
+    },
+    {
+      sqlTexts:[
+        'select * from db1.public.t1 x, db2.public.t2 y, db3.public.t3 z where x.a = y.a and y.a = z.a;',
+        'select * from db1.public.t1 x, db2.public.t2 y where x.a = y.a;',
+        'select * from db2.public.t2 y, db3.public.t3 z where y.a = z.a;'
+      ],
+      QccSize:4,
+    },
+  ];
+
+  function createQueryTest () {
+    const testingSet = [];
+    for(let i = 0; i < querySet.length; i++) {
+      const testingFunction = function(callback) {
+        const {sqlTexts,QccSize} = querySet[i];
+        connection.execute({
+          sqlText: sqlTexts[0],
+          complete: function (err) {
+            if (err) {
+              callback(err);
+            }
+          }
+        });
+        connection.execute({
+          sqlText: sqlTexts[1],
+          complete: function (err) {
+            if (err) {
+              callback(err);
+            }
+          }
+        });
+        connection.execute({
+          sqlText: sqlTexts[2],
+          complete: function (err, stmt) {
+            assert.ok(!err,'There should be no error!');
+            assert.strictEqual(stmt.getQueryContextCacheSize(), QccSize);
+            assert.strictEqual(stmt.getQueryContextDTOSize(),QccSize);
+            callback(); 
+          }
+        });
+      };
+      testingSet.push(testingFunction);
+    }
+    return testingSet;
+  }
+
+  it('test Query Context Cache', function (done) {
```

Comment:
I ran this test and it looks like there is some race problem (I added some additional logs where checking errors):

```
  Query Context Cache test
Running query create or replace database db1
Error is null
Rows are [{"status":"Database DB1 successfully created."}]
Running query insert into t1 values (1, 2), (2, 3), (3, 4)
Error is null
Rows are [{"number of rows inserted":3}]
Running query create or replace hybrid table t1 (a int primary key, b int)
Error is null
Rows are [{"status":"Table T1 successfully created."}]
Running query create or replace database db2
Error is null
Rows are [{"status":"Database DB2 successfully created."}]
Running query insert into t2 values (1, 2), (2, 3), (3, 4)
Error is {"name":"OperationFailedError","code":"002003","message":"SQL compilation error:\nTable 'T2' does not exist or not authorized.","sqlState":"42S02","data":{"internalError":false,"errorCode":"002003","age":0,"sqlState":"42S02","queryId":"01af166b-0000-9308-0000-01110003713a","line":1,"pos":12,"type":"COMPILATION"}}
Rows are undefined
    1) test Query Context Cache


  0 passing (4s)
  1 failing

  1) Query Context Cache test
       test Query Context Cache:

      Uncaught AssertionError [ERR_ASSERTION]: There should be no error!
      + expected - actual

      -false
      +true

Running query create or replace hybrid table t2 (a int primary key, b int)
Error is null
Rows are [{"status":"Table T2 successfully created."}]
```

I think it's because for queryTests `callback` function is not properly called